torch/masked/_ops.py

```diff
@@ -1386,6 +1386,10 @@ def mean(
 {reduction_example}"""
     if dtype is None:
         dtype = input.dtype
+    if dtype is torch.bool:
+        # upcast dtype when computing mean of boolean tensors otherwise
+        # sum(..., dtype=torch.bool) is clamped to 1
+        dtype = torch.int32
```

Comment:
Hm, technically the correct result would be to throw an exception

Reply:
```
>>> torch.randn(40, 50).bool().mean()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: mean(): could not infer output dtype. Input dtype must be either a floating point or complex dtype. Got: Bool
```

Reply:
I've also changed the supported dtypes for masked.mean to match the supported dtypes for mean.